### PR TITLE
retry on connection error to neutron

### PR DESCRIPTION
**Pull request by kpawar-sap**
- create port
- get network
- show port

### CODE REVIEW

**Comment by kpawar-sap** [comment]
https://github.com/sapcc/manila/issues/138 issue

File: manila/network/neutron/api.py [hunk 1]
```diff
@@ -167,6 +182,15 @@ def create_port(self, tenant_id, network_id, host_id=None, subnet_id=None,
                 raise exception.PortLimitExceeded()
             raise exception.NetworkException(code=e.status_code,
                                              message=e.message)
+        except ks_exec.ConnectFailure:
+            LOG.warning('Create Port: Neutron connection failure')
+            # check if port is created in neutron else re-raise connectFailure
+            search_opts = {'device_id': device_id, 'network_id': network_id}
```
Comment (Carthaca) [inline manila/network/neutron/api.py:188]: Sadly this combination is not enough, we have `get_network_allocations_number()` (in the netapp driver this translates to the number of cluster nodes, usually 2 in our landscape) times ports with the same device_id + network_id combination.
Maybe we should start giving the ports different `name`s, just counting up the allocation numbers to have something unique for identification?
I.e. I imagine the caller of this method `def allocate_network()` to pass a name containing the `allocation_count`
WDYT?

**Review by Carthaca — CHANGES_REQUESTED** [review]

File: manila/network/neutron/api.py (hunk 1 above)
Comment (kpawar-sap) [reply manila/network/neutron/api.py]: updated

**Review by Carthaca — APPROVED** [review]

File: manila/network/neutron/api.py (hunk 1 above)
Comment (Carthaca) [reply manila/network/neutron/api.py]: thanks

**Pull request merged**